cloudimg/ms_azure.py

```diff
@@ -406,8 +419,17 @@ def upload_to_container(self, image_path, container_name, object_name,
         if image_path.startswith("http:") or image_path.startswith("https:"):
             blob_client.upload_blob_from_url(source_url=image_path, **kwargs)
         else:
-            with open(image_path, "rb") as data:
-                blob_client.upload_blob(data=data, **kwargs)
+            with open_func(image_path, "rb") as data:
+                # we need to pass the lenght to upload_blob for optimizaion and
+                # because without it the LZMA won't decompress correctly.
+                bytes_count = data.seek(0, os.SEEK_END)
```

Comment:
Hmm, I don't have another solution to offer but this is rather unfortunate.  I think this means a compressed file is being decompressed twice, once just to get the length and then again as it's uploaded.  There's probably no way around it.

Reply:
Yep; it is not ideal. But there is not any other way of getting size for xz at the moment.

ATM we have jsut 2 ways how to handle upload of xz files with azure-storage-blob
* current one with lzma.open and decompressing it twice to get length
* decompress it temporary directory.

@rohanpm Each of them has some pros and cons. I tested and getting size of xz file with `data.seek(0, os.SEEK_END)`
takes the similar time as with `xz --test`. It is approximately extra 3 minutes for 64 GiB after decompression.